java/arcs/sdk/android/storage/service/StorageService.kt

```diff
@@ -66,6 +68,9 @@ open class StorageService : ResurrectorService() {
             Thread(it).apply { name = "WriteBack #$id" }
         }.asCoroutineDispatcher() + SupervisorJob()
     )
+
+    private val cleanupScope by lazy { CoroutineScope(coroutineContext) }
```

Comment:
I think both the writeback scope and cleanup scope could both be child scopes of a single parent as long as it's okay to cancel one if the other fails for some reason.

Reply:
I could use more guidance here.

One tangential comment... I needed the lazy constructor here because `coroutineContext` can be overridden, so we can't construct the scope at object construction time.

Maybe there's a bet structure for all of this, though?

Reply:
I don't understand how this is equivalent to what we had. Here you are cleaning in onUnbind, whereas before you were cleaning up when there were no callbacks left.
They are not equivalent: servicestore.off removes the callback but it does not unbind.
and it looks like we call off each time we stop an arc, but we never unbind until lifecycle destroyed.

Reply:
Oh darn, you're right. I was thinking we also closed the ServiceStores along with the arc, but that's not true. :-(

Reply:
however, ,maybe it is better to change that: ie when servicestore.off is called, it should unbind. I don't see why we keep the binding active.

Reply:
I have a separate thread of work that's investigating just this question :-)

Reply:
that's music to my ears :)